From 14fb6f0a31373b242bdeb18cf9f3a36441bf86b5 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Tue, 3 Jan 2023 16:55:59 +0100
Subject: [PATCH] Add support for ZXing 2.0

As far as we are concerned that is source-compatible, so nothing we need
to change here apart from adjusting the build system to accept the new
major version.
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 365a1c9..961262e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,7 +44,10 @@ find_package(Dmtx)
 set_package_properties(Dmtx PROPERTIES
     PURPOSE "Required for generation of Data Matrix barcodes."
     TYPE RECOMMENDED)
-find_package(ZXing 1.2.0)
+find_package(ZXing 2.0)
+if (NOT TARGET ZXing::ZXing)
+  find_package(ZXing 1.2.0)
+endif()
 set_package_properties(ZXing PROPERTIES
     PURPOSE "Required for generation of PDF417 barcodes and for scanning of barcodes from live video feed."
     TYPE RECOMMENDED)
-- 
GitLab

